### PR TITLE
HDDS-7361. Add general metrics for queues in Datanode

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

This ticket is to add a general metrics class for the following queues in Datanodes

* StateContextCommandQueue
* CommandDispatcherQueue
* IncrementalReportQueue
* ContainerActionQueue
* PipelineActionQueue

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7361

## How was this patch tested?

unit test.

### CODE REVIEW

**Comment by symious** [comment]
@sodonnel Could you help to review this PR?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeQueueMetrics.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            CaseFormat.UPPER_UNDERSCORE
+                .to(CaseFormat.UPPER_CAMEL, k.getHostName())));
+    pipelineActionQueueMap.computeIfAbsent(endpoint,
+        k -> getMetricsInfo(PIPELINE_ACTION_QUEUE_PREFIX,
+            CaseFormat.UPPER_UNDERSCORE
+                .to(CaseFormat.UPPER_CAMEL, k.getHostName())));
+  }
+
+  private MetricsInfo getMetricsInfo(String prefix, String metricName) {
+    String metric = prefix + WordUtils.capitalize(metricName);
```
Comment (sodonnel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeQueueMetrics.java:175]: Do you think we should append count / size / total to the end of the metric name? Most counters seem to end in something like this.

**Comment by symious** [comment]
@sodonnel Added "size" to metrics names, please have a look. 
I'm not sure if we need to change the order of the metrics name or not.

**Review by sodonnel — APPROVED** [review]
Thanks for taking on the changes. LGTM!

**Pull request merged**